OpenLaTeXStudio/Editor/src/main/java/latexstudio/editor/EditorTopComponent.java

```diff
@@ -276,6 +287,51 @@ public void commentOutText() {
         }
 
     }
+    
+    public void spellCheckAllText() throws BadLocationException {     
+        Document doc = rSyntaxTextArea.getDocument();   
+        if (doc != null) {                                                
+            String editorText = rSyntaxTextArea.getText(0, doc.getLength());            
+            if (editorText != null) {  
+                Highlighter highlighter = rSyntaxTextArea.getHighlighter();
+                Highlighter.HighlightPainter painter = new DefaultHighlighter.DefaultHighlightPainter(Color.PINK);  //Default color is: PINK
+
+                JLanguageTool langTool = new JLanguageTool(new AmericanEnglish());  //Default language is: American English
+                for (Rule rule : langTool.getAllActiveRules()) {
+                    if (rule instanceof SpellingCheckRule) {
+                        ((SpellingCheckRule)rule).acceptPhrases(Arrays.asList("tex", "Tex", "TEX", "documentclass", "maketitle"));  //Accept some TEX terms

```

Comment:
We've got they key LaTeX terms in a resource file tex.cwl that is used for auto-complete. We could also use it to add exception phrases here, but this is optional for now.